tensorflow_datasets/core/registered.py

```diff
@@ -106,6 +107,14 @@
     version=r"[0-9]+\.[0-9]+\.[0-9]+",
 ))
 
+_skip_reg = False
+
+@contextmanager
+def skip_registration():
+  global _skip_reg
+  _skip_reg = True
+  yield
```

Comment:
It is good practice to wraps yield inside a try/finally block, to execute `_skip_reg = False` even in case an exception is raised in the context manager.